src/main/java/org/team1540/robot2018/Robot.java

```diff
@@ -376,4 +204,199 @@ public void teleopPeriodic() {
+    LEFT_HOOK_SWITCH_THEN_SCALE("Left Hook Switch Then Scale", new DecisionNode(new DecisionNode[]{
+        new DecisionNode(Posession.SWITCH_OWNED_LEFT, MotionProfile.LEFT_HOOK),
+        new DecisionNode(Posession.SCALE_OWNED_LEFT, MotionProfile.LEFT_SCALE_STRAIGHT),
+        new DecisionNode(Posession.SCALE_OWNED_RIGHT, MotionProfile.GO_STRAIGHT),
+        new DecisionNode(Posession.NO_DATA, MotionProfile.GO_STRAIGHT).setMessage(new Message("Could not get match data, reverting to base auto", true))
+    })),
+    MIDDLE("Middle", new DecisionNode(new DecisionNode[]{
+        new DecisionNode(Posession.SWITCH_OWNED_LEFT, MotionProfile.MIDDLE_TO_LEFT_SWITCH),
+        new DecisionNode(Posession.SWITCH_OWNED_RIGHT, MotionProfile.MIDDLE_TO_RIGHT_SWITCH),
+        new DecisionNode(Posession.NO_DATA, MotionProfile.GO_STRAIGHT).setMessage(new Message("Could not get match data, reverting to base auto", true))
+    })),
+    CENTER_DOUBLE_CUBE("Center Double Cube", new DecisionNode(new DecisionNode[]{
+       new DecisionNode(Posession.SWITCH_OWNED_LEFT, MotionProfile.SWITCH_DOUBLE_CUBE_LEFT),
+       new DecisionNode(Posession.SWITCH_OWNED_RIGHT, MotionProfile.SWITCH_DOUBLE_CUBE_RIGHT)
+    })),
+    RIGHT_HOOK_SWITCH("Right Hook Switch", new DecisionNode(new DecisionNode[]{
+        new DecisionNode(Posession.SWITCH_OWNED_RIGHT, MotionProfile.RIGHT_HOOK),
+        new DecisionNode(Posession.NO_DATA, MotionProfile.GO_STRAIGHT)
+    })),
+    RIGHT_SCALE_THEN_SWITCH("Right Scale Then Switch", new DecisionNode(new DecisionNode[]{
+        new DecisionNode(Posession.SCALE_OWNED_RIGHT, MotionProfile.RIGHT_SCALE),
+        new DecisionNode(Posession.SCALE_OWNED_LEFT, new DecisionNode[]{
+            new DecisionNode(Posession.SCALE_OWNED_RIGHT, MotionProfile.RIGHT_HOOK),
+            new DecisionNode(Posession.SWITCH_OWNED_LEFT, MotionProfile.GO_STRAIGHT),
+            new DecisionNode(Posession.NO_DATA, MotionProfile.GO_STRAIGHT).setMessage(new Message("Could not get match data, reverting to base auto", true))
+        }),
+        new DecisionNode(Posession.NO_DATA, MotionProfile.GO_STRAIGHT).setMessage(new Message("Could not get match data, reverting to base auto", true))
+    })),
+    CROSS_LINE("Cross Line", new DecisionNode(MotionProfile.GO_STRAIGHT)),
+    STUPID("Stupid", new DecisionNode(MotionProfile.DRIVE_TIMED));
+
+
+    private final String name;
+    private final DecisionNode root;
+
+    AutoMode(String name, DecisionNode root) {
+      this.name = name;
+      this.root = root;
+    }
+
+  }
+
+  private enum Posession {
+
+    SCALE_OWNED_LEFT(() -> (MatchData.getOwnedSide(GameFeature.SCALE) == OwnedSide.LEFT)),
+    SCALE_OWNED_RIGHT(() -> (MatchData.getOwnedSide(GameFeature.SCALE) == OwnedSide.RIGHT)),
+    SWITCH_OWNED_LEFT(() -> (MatchData.getOwnedSide(GameFeature.SWITCH_NEAR) == OwnedSide.LEFT)),
+    SWITCH_OWNED_RIGHT(() -> (MatchData.getOwnedSide(GameFeature.SWITCH_NEAR) == OwnedSide.RIGHT)),
+    NO_DATA(() -> (true));
```

Comment:
This shouldn't always be true, it should actually check if `getOwnedSide()` returns `UNKNOWN`